### ISSUE TITLE
What tag is rendering this building as dark brown?

### ISSUE THREAD

**Issue by DaveF63** [gravitystorm/openstreetmap-carto#3257]
https://www.openstreetmap.org/way/41599363

Given the tags I thought it should be the standard building colour, but it's dark, as if it contains the tag building=train_station, but Overpass doesn't return one.

I notice 'public_transport=station' is included with building=train_station in buildings.mss under #buildings-major but the examples I've seen render that as a lighter colour when it's a polygon.

**Comment by matthijsmelissen** [comment]
It is indeed public_transport=station. 

>  the examples I've seen render that as a lighter colour when it's a polygon.

If that is the case that seems to be a but, for example https://www.openstreetmap.org/way/299305320 renders like it should. Let me know if you have a counterexample.

**Comment by DaveF63** [comment]
OK, For some reason that renders dark brown as its combined with building=yes. 
Yet carto doesn't do similar when combined with railway=station. Note the difference in the two Shepherd's Bush stations. https://www.openstreetmap.org/#map=19/51.50494/-0.21840

public_transport=station & railway=station are comparable: https://wiki.openstreetmap.org/wiki/Tag%3Apublic_transport%3Dstation

However I don't see them as relating to a building. They represent the whole of station. There should only be one tag per station, but often stations will contain multiple buildings. This leads to inconsistency in how station buildings will be tagged. I think they should have the darker render if tagged building=train_station 
OK, For some reason that renders dark brown as its combined with building=yes. 
Yet carto doesn't do similar when combined with railway=station. Note the difference in the two Shepherd's Bush stations. https://www.openstreetmap.org/#map=19/51.50494/-0.21840

`public_transport=station` & `railway=station` are comparable: https://wiki.openstreetmap.org/wiki/Tag%3Apublic_transport%3Dstation

However I don't see them as relating to a building. They represent the whole of station. There should only be one tag per station, but often stations will contain multiple buildings. This leads to inconsistency in how station buildings will be tagged. I think they should have the darker render only if tagged `building=train_station `

**Comment by matthijsmelissen** [comment]
Reopening - although it is not yes clear to me exactly what your suggested solution would be.

**Comment by HolgerJeromin** [comment]
@matthijsmelissen forgot to reopen?

**Comment by kocio-pl** [comment]
I guess it might be a tagging scheme problem.

**Comment by jeisenbe** [comment]
There are 2 problems here:

1) `public_transport=station` + `building` is rendered is the dark brown major_buildings color, but not the equivalent tags `railway=station` or `amenity=bus_station`, ferry terminal, aerialway station, etc - we should either drop the " major building" rendering for public_transport=station, or add the others. 

I would recommend dropping rendering for `public_transport=station`, since normally the station should either be just a node, or a larger area that is not identical to the building.

2) `building=train_station` is rendered as a major building, but not other public transport building types.

**Comment by kocio-pl** [comment]
> normally the station should either be just a node, or a larger area that is not identical to the building.

I don't see justification for that claim. For example `amenity=school` can be any of them.

**Comment by jeisenbe** [comment]
It certainly can be tagged identically to a building, but we shouldn't
necessarily encourage or endorse such tagging as superior to the other
options. Giving a special rendering to buildings with
public=transport_station implies that this tag should be added to the
building area.

**Comment by kocio-pl** [comment]
On the other hand we do special rendering for airport terminals and I see that makes sense.

**Comment by jeisenbe** [comment]
Yes, but the airport terminal is defined as a building, while a train
or bus station often includes areas of tracks and platforms which are
not always inside of a building.

**Comment by kocio-pl** [comment]
Do you see (or envision) some other tagging that could be used as a universal indicator of terminal building?

**Comment by jeisenbe** [comment]
I'm sorry if I wasn't clear. 

I think that it is good to check for `aeroway=terminal` plus `building` and use that for rendering terminal uniquely, as we currently do, because an `aeroway=terminal` is always a building, [by definition](https://wiki.openstreetmap.org/wiki/Tag:aeroway=terminal): "an aeroway=terminal is a building... .

In contrast: "[railway=station](https://wiki.openstreetmap.org/wiki/Tag:railway%3Dstation) should be applied to an area covering the whole station, or preliminary on a node placed at the center of the station area" - so this is usually not a building, but either a node or a larger area.

For [amenity=bus_station](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbus_station): "A bus station consists of several bus stops, usually allowing people to travel in many different directions" - no mention of a building, and almost 2/3rds are mapped as nodes.

public_transport=station: " If the station is completely (or most of it) inside a station building, the building is the outline. If the station is not fully or mostly covered by a building, any existing buildings should be mapped at their place inside the station outline." - 42k mapped as nodes, only 17k as ways, only 12k combined with `building=`.

Since mappers do not usually add the station tags to a building feature, we should not rely on this for rendering. It gives confusing feedback to mappers, and usually doesn't work.

**Comment by kocio-pl** [comment]
No need to be sorry, I understand you. 

I just asked if you can see the universal solution for this, because you're also very active with tagging documentation and maybe you have some good idea to solve it.
